ecs-agent/acs/session/session.go

```diff
@@ -234,7 +238,14 @@ func (s *session) Start(ctx context.Context) error {
 // startSessionOnce creates a session with ACS and handles requests using the passed
 // in arguments.
 func (s *session) startSessionOnce(ctx context.Context) error {
+	if s.GetFirstDiscoverPollEndpointTime().IsZero() {
+		s.firstDiscoverPollEndpointTime = time.Now()
+	}
+
+	discoverPollEndpointMetric := s.metricsFactory.New(metrics.ACSDiscoverPollEndpointDurationName)
```

Comment:
The DiscoverPollEndpoint can be called by TACS session as well. Can we emit the metric in this single place - https://github.com/aws/amazon-ecs-agent/blob/master/ecs-agent/api/ecs/client/ecs_client.go#L679? which is used by both ACS and TACS.

Reply:
good point, will move the metric over there